lib/core_ext/numpy.rb

```diff
@@ -8,6 +8,20 @@ def d2!
     raise 'not a 2d array' if !first.is_a?(Array)
   end
 
+  def pts!(x1, y1, x2, y2)
+    if x1 < 0 || y1 < 0 || x2 >= self[0].length || y2 >= length
+      raise ArgumentError, "must be in bounds. 0 < #{x1} < #{x2} < #{first.length} and 0 < #{y1} < #{y2} < #{length}"
+    end
+
+    if x1 > x2 || y1 > y2
+      raise ArgumentError, 'first point must be before the second point'
+    end
+  end
+
+  def sum_all
+    sum { |row| row.sum }
+  end
+
```

Comment:
use `_1`. Is this a numpy method, or just a helper that you like? Consider renaming to sum_2d. Also, add a comment